### ISSUE TITLE
Bug: incorrect type definitions in `createResolveProtocol` function

### ISSUE THREAD

**Issue by shellvon** [szmarczak/http2-wrapper#107]
Currently, the default cache type in the codebase is `QuickLRU<string, string>`, as seen in the implementation at [this line](https://github.com/szmarczak/http2-wrapper/blob/f1a1776d4b1aef1cd47d55b1a950986e86145a09/source/auto.js#L14) and [this line](https://github.com/szmarczak/http2-wrapper/blob/f1a1776d4b1aef1cd47d55b1a950986e86145a09/source/auto.js#L96). However, according to the type definitions specified in the TypeScript declarations file [here](https://github.com/szmarczak/http2-wrapper/blob/f1a1776d4b1aef1cd47d55b1a950986e86145a09/index.d.ts#L133), the expected type for the cache is `Map<string, string>`.

When attempting to use this library, TypeScript reports a type mismatch error, likely because the provided type (`QuickLRU<string, string>`) does not match the expected type (`Map<string, string>`).

report example:

![image](https://github.com/szmarczak/http2-wrapper/assets/3349696/2ab9ca07-a0cf-475a-a1d0-18e32afeee41)

Given the current implementation, the cache type here can be any object that implements the following interface:
```typescript
interface Cache {
    has(k: string): boolean;
    get(k: string): string | undefined;
}
```
In other words, `cache` can be any object instance that has both `has` and `get` methods defined.
